hyperliquid/utils/types.py

```diff
@@ -29,6 +29,14 @@
 )
 SpotMeta = TypedDict("SpotMeta", {"universe": List[SpotAssetInfo], "tokens": List[SpotTokenInfo]})
 
+
+SpotdAssetCtxsTokenInfo = TypedDict("SpotdAssetCtxsTokenInfo", {"name": str, "szDecimals": int, "weiDecimals": int})
+SpotdAssetCtxsUniverseInfo = TypedDict("SpotdAssetCtxsUniverseInfo", {"name": str, "tokens": List[int]})
+SpotdAssetCtxsMarketDataInfo = TypedDict("SpotdAssetCtxsMarketDataInfo", {"dayNtlVlm": str, "markPx": str, "midPx": str, "prevDayPx": str})
+
+SpotMetaAndAssetCtxsDict = TypedDict("SpotMetaAndAssetCtxsDict", {"tokens": List[SpotdAssetCtxsTokenInfo], "universe": List[SpotdAssetCtxsUniverseInfo]})
+SpotMetaAndAssetCtxs = List[Union[SpotMetaAndAssetCtxsDict, List[SpotdAssetCtxsMarketDataInfo]]]
```

Comment:
This type can be better, it should just be a tuple of SpotMeta and List[SpotAssetCtx]:
SpotMetaAndAssetCtxs = Tuple[SpotMeta, List[SpotAssetCtx]]

Reply:
well, I tried to replicate the response which is 
array[json{},array[]]
```
[
{
    "tokens": [
        {
...
        },
        {
...
        }
    ],
    "universe": [
        {
...
        }
    ]
},
[
    {
...
    }
]
]
```

Reply:
The top level result should always have 2 elements the first of which is the SpotMeta and the second one is a list of SpotAssetCtx, if you define it as a list of a union that doesn't specify anything about the length or order of the response

Reply:
u r right, I implemented the tuple.